internal/datastore/postgres/migrations/zz_migration.0009_add_xid8_columns.go

```diff
@@ -0,0 +1,46 @@
+package migrations
+
+import (
+	"context"
+
+	"github.com/jackc/pgx/v4"
+)
+
+const addTransactionXIDColumns = `
+	ALTER TABLE relation_tuple_transaction
+		ADD COLUMN xid xid8 NOT NULL DEFAULT (pg_current_xact_id()),
+		ADD COLUMN snapshot pg_snapshot NOT NULL DEFAULT (pg_current_snapshot())`
+
+const addTupleXIDColumns = `
+	ALTER TABLE relation_tuple
+		ADD COLUMN created_xid xid8 NOT NULL DEFAULT (pg_current_xact_id()),
+		ADD COLUMN deleted_xid xid8 NOT NULL DEFAULT ('9223372036854775807');`
```

Comment:
I'm curious, why creating those when they are available as metadata at the row level that can be queried? And why using `deleted_xid` as the max value when PSQL stores it as `0` when it's not yet deleted?

Reply:
`xmin` is of type `xid` which is only 32 bits, the `pg_currext_xact_id()` is and `xid8` which is 64-bits and isn't subject to the transaction rollover problem. It's also the data-type that's needed for us with the `pg_snapshot` functions. You can cast an `xid8` to an `xid`, for example with the `pg_xact_commit_timestamp` function but not vice-versa.

Originally we used the max int64 because the conditionals were simpler, e.g. `<=`, but since they can be out of order we have to compare with the actual value so the utility isn't as obvious. It's still an `exceptional` value either way. The only reason I can see to keep it "very high" is because I believe that `pg_visible_in_snapshot` may not special case 0 and it will return true when the `xmin` gets higher.